presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -490,11 +492,19 @@ protected Scope visitTable(Table table, Scope scope)
 
         Optional<ViewDefinition> optionalView = metadata.getView(session, name);
         if (optionalView.isPresent()) {
+            Statement statement = analysis.getStatement();
+            if (statement instanceof CreateView && ((CreateView) statement).isReplace() && ((CreateView) statement).getName().equals(table.getName())) {
+                throw new SemanticException(VIEW_CANNOT_BE_RECURSIVE, table, "Statement would create a recursive view.");
+            }
+            else if (analysis.hasTableInView(table)) {
```

Comment:
I'm not sure whether this will create false positives. Another way is to record a graph and check for loop, but that would be more expensive.